### PR TITLE
Add set_Lagrangian in Wavefunction

### PR DESCRIPTION
**Pull request by lcyyork**
## Description
Add set_Lagrangian in Wavefunction. Also expose `C_DGEQP3` function. 

## Checklist
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests) `ctest -L quick` all passed.

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

**Review by loriab — APPROVED** [review]
lgtm!

add a bump commit after https://github.com/psi4/psi4/pull/2065 merged, and CI should pass. sorry about that.

**Comment by lcyyork** [comment]
Thank you @loriab. This just contains some random changes that will make our gradient implementation easier in Forte.

**Review by PeterKraus — APPROVED** [review]

File: psi4/src/psi4/libmints/wavefunction.h [hunk 1]
```diff
@@ -574,8 +572,12 @@ class PSI_API Wavefunction : public std::enable_shared_from_this<Wavefunction> {
     SharedMatrix basis_projection(SharedMatrix Cold, Dimension noccpi, std::shared_ptr<BasisSet> old_basis,
                                   std::shared_ptr<BasisSet> new_basis);
 
+    /// Returns the SO basis Lagrangian
```
Comment (dgasmith) [inline psi4/src/psi4/libmints/wavefunction.h:575]: Should be lowercase I believe?

**Review by dgasmith — APPROVED** [review]

File: psi4/src/psi4/libmints/wavefunction.h (hunk 1 above)
Comment (andysim) [reply psi4/src/psi4/libmints/wavefunction.h:575]: I think I agree.  It's [correct to capitalize terms derived from names](http://folk.ntnu.no/haugwarb/Misc/capitalization.pdf) in writing, but I think we want to be as consistent in the API as we can.  An example of the is is the `aotoso` function directly below; here AO and SO should normally be capitalized, but are left lower case.

Comment (lcyyork) [reply psi4/src/psi4/libmints/wavefunction.h:575]: I rename the functions to lowercase, though I did not touch the variable name (still `Lagrangian_`).

Comment (andysim) [reply psi4/src/psi4/libmints/wavefunction.h:575]: Thanks @lcyyork - the variable names can be capitalized any way, as long as the API is consistent, which it is now

**Comment by loriab** [comment]
rebase this, please. then, if Azure triggers, this should be good to merge.

**Comment by loriab** [comment]
whoops, we forgot that this PR was adding a setter to accompany an existing getter, not exposing afresh, so a change in case was not appropriate. as usual, v2rdm found the trouble:

```
/Users/github/builds/conda-builds/v2rdm_casscf_1608103337174/work/v2rdm_solver.cc:420:57: error: no member named 'Lagrangian' in 'psi::Wavefunction'; did you mean 'lagrangian'?
    Lagrangian_ = SharedMatrix(reference_wavefunction_->Lagrangian());
                                                        ^~~~~~~~~~
                                                        lagrangian
/Users/github/builds/conda-builds/v2rdm_casscf_1608103337174/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include/psi4/libmints/wavefunction.h:576:18: note: 'lagrangian' declared here
    SharedMatrix lagrangian() const;
                 ^
```

Please change back to caps C-side. I'm inclined to have Py-side match C-side, but there's leeway for the export_ files. Sorry for the zigzag advice, @lcyyork.

**Comment by lcyyork** [comment]
Thanks @loriab, changes have been made. Please let me know if I should change the Py-side as well.

We do have a duplicate in the wavefunction where `Lagrangian()` and `X()` return the same thing. Let me know if I should remove one of those.

**Pull request merged**